src/host/VtIo.cpp

```diff
@@ -167,49 +154,49 @@ bool VtIo::IsUsingVt() const
     {
         return S_FALSE;
     }
-    auto& g = ServiceLocator::LocateGlobals();
 
-    if (_pVtRenderEngine)
+    if (_pVtInputThread)
     {
-        try
-        {
-            g.pRender->AddRenderEngine(_pVtRenderEngine.get());
-            g.getConsoleInformation().GetActiveOutputBuffer().SetTerminalConnection(_pVtRenderEngine.get());
-
-            // Force the whole window to be put together first.
-            // We don't really need the handle, we just want to leverage the setup steps.
-            ServiceLocator::LocatePseudoWindow();
-        }
-        CATCH_RETURN();
+        LOG_IF_FAILED(_pVtInputThread->Start());
     }
 
-    if (_pVtInputThread)
     {
-        LOG_IF_FAILED(_pVtInputThread->Start());
+        auto writer = GetWriter();
+
+        // GH#4999 - Send a sequence to the connected terminal to request
+        // win32-input-mode from them. This will enable the connected terminal to
+        // send us full INPUT_RECORDs as input. If the terminal doesn't understand
+        // this sequence, it'll just ignore it.
+
+        writer.WriteUTF8(
+            "\033[?1004h" // Focus Event Mode
+            "\033[?9001h" // Win32 Input Mode

```

Comment:
There was an ordering bug that resulted in a deadlock if conhost (inbox) sent 9001h **before requesting CPR** since it might receive it in an unusable format. How does this work now?

Reply:
I solved the double encoding of VT sequences with WIN32IM encoding a while back so this shouldn't happen anymore. If I'm mistaken and it still happens, we should definitely fix the double-encoding IMO, because it makes no sense that a VT response would be encoded via WIN32IM.